From 7403fcc4f07c9bd72a6b3994c87ceb023f974a24 Mon Sep 17 00:00:00 2001
From: Artem Lapkin <art@khadas.com>
Date: Thu, 18 Nov 2021 10:22:04 +0800
Subject: [PATCH v1 2/2] PCI: DWC: meson: setup 512 PCIe Max Read Request Size

DWC PCIe controller imposes limits on the Read Request Size that it can
handle.

For pci-meson family it's fixed at 512 bytes by default.

Signed-off-by: Artem Lapkin <art@khadas.com>
---
 drivers/pci/controller/dwc/pci-meson.c | 1 +
 1 file changed, 1 insertion(+)

V1: remake old patches
https://lkml.org/lkml/2021/7/28/1237
https://patchwork.kernel.org/project/linux-pci/patch/m3lf36n0d8.fsf@t19.piap.pl/

diff --git a/drivers/pci/controller/dwc/pci-meson.c b/drivers/pci/controller/dwc/pci-meson.c
index 686ded034f22..f0b02d2c3e96 100644
--- a/drivers/pci/controller/dwc/pci-meson.c
+++ b/drivers/pci/controller/dwc/pci-meson.c
@@ -451,6 +451,7 @@ static int meson_pcie_probe(struct platform_device *pdev)
 		goto err_phy;
 	}
 
+	max_pcie_mrrs = 512;
 	platform_set_drvdata(pdev, mp);
 
 	ret = dw_pcie_host_init(&pci->pp);
-- 
2.25.1

